types/src/lib.rs

```diff
@@ -5,17 +5,20 @@
 
 extern crate alloc;
 
-/// JSON-RPC 2.0 specification related types.
-pub mod jsonrpc;
-
 /// JSON-RPC 2.0 specification related types v2.
 pub mod v2;
 
 /// Shared error type.
 pub mod error;
 
 /// Shared client types.
-pub mod client;
+mod client;
 
 /// Traits
 pub mod traits;
+
+pub use beef::Cow;
+pub use client::*;
+pub use error::Error;
```

Comment:
we could revisit this error type but it's quite hard to understand the different error types.

Technically we could kill the huge enum and have each client/server to have their own error type instead.